src/web_ui/__init__.py

```diff
@@ -226,7 +226,7 @@ def wifi_ap_status():
 
 
 def test_online():
-    TEST_URL = 'http://www.google.com/'
+    TEST_URL = 'http://edge.ammp.io/'
```

Comment:
Hm let's make 100% sure this will actually behave as intended. If I run
```
curl http://edge.ammp.io/
```
I get a 301 response (HTTPS redirect) which is fine.

But for some reason on my local machine if I run (in Python)
```
from urllib.request import urlopen
urlopen('http://edge.ammp.io/')
```
I get a 403 Forbidden. I guess that's because it follows the 301 redirect? What do you get?

Reply:
Yeah just verified this. The `urlopen()` function follows the redirect, establishes an HTTPS connection, and then gets a 403 forbidden - which is an exception.

Testing against a resource that returns 403 is a bit of an iffy one, but it can be addressed by checking for a lower-level connection:
```
import socket
conn = socket.create_connection(('edge.ammp.io', 80), timeout=15)
```
This will simply check whether a TCP connection can be established on the given port. If it's possible, it stands to reason that the datalogger is online (we don't really need to do "proper" communication over HTTP as further proof). If it's not possible it will raise an exception.

If no exception is raised then we should be nice and do `conn.close()` before moving on, in order to not leave a dangling open connection.

Reply:
In this context we just need the hostname (edge.ammp.io) rather than a full URL, so we should also edit the variable names accordingly (`TEST_HOST`, `TEST_PORT`; or just `TEST_HOST_PORT = ('edge.ammp.io', 80)`)

Reply:
thanks for checking! i tried to ping inside the datalogger to just the host "edge.ammp.io" before and that had similar as pinging google.com, so I didn't check the behavior of the code with `urlopen`. Think I get the points you mentioned, will test it now

Reply:
> 

I actually get an exception:
```
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)
```
but not the 403, maybe I missed some steps?

Reply:
ok nvm I get 403 now, i was running python3 outside of the dev environment so I didn't have the SSL cert